MIDI: support port name strings

kaysievers opened the pull request:
# MIDI port names

This patch allows to set custom descriptive names for virtual MIDI ports/cables. A MIDI port offers different functionality on the same physical USB device. The names make it easier to distinguish one from the other.

Edit: the tinyusb part was merged: https://github.com/hathach/tinyusb/pull/1738

# Screenshot of Chrome on MacOS listing WebMIDI devices:
<img width="548" alt="Screenshot 2022-11-10 at 17 03 00" src="https://user-images.githubusercontent.com/1064936/201146950-028ab1c3-7c2b-4239-8929-02f7130d8194.png">

# Screenshot of MacOS system-level MIDI Setup
<img width="639" alt="Screenshot 2022-11-10 at 17 02 35" src="https://user-images.githubusercontent.com/1064936/201146906-d052aa1d-9832-4418-9703-5fcf9280c995.png">

# Example calls in setup() of Arduino sketch:

```c
  Device.usb.midi.setPortName(1, "Keyboard");
  Device.usb.midi.setPortName(2, "Drum Pads");
  Device.usb.midi.setPortName(3, "Lights");
```

# Calls to Adafruit_TinyUSB_Arduino from a local/custom MIDI Library:

```c
  void setPorts(uint8_t n_ports) {
    _midi.interface.setCables(n_ports);
  }

  void setPortName(uint8_t port, const char *name) {
    _midi.interface.setCableNameStringIndex(port, _device.device->addStringToIndex(name));
  }
```

hathach commented:
tinyusb core is sync with upstream, could you revert changes in usbd.h and merge from master to update PR. Note: the pr may still fail on esp32 ci since esp32 update its tinyusb core separately, which requires a bit of time for it to sync up as well.

kaysievers commented:
> could you revert changes in usbd.h and merge from master

I've updated the PR. Thanks!

kaysievers commented:
@hathach is there anything I can do to get this merged? Thanks!

hathach commented:
> @hathach is there anything I can do to get this merged? Thanks!

nothing from your side, it is my issue. I will try to review and test this out asap.

hathach reviewed (approved):
Thank you very much for the PR, I also made some changes to the PR, midi setCableName() take string and add then save the string index internally, that makes API look cleaner.  Will merge when ci passed.

```C
  usb_midi.setCableName(1, "Keyboard");
  usb_midi.setCableName(2, "Drum Pads");
  usb_midi.setCableName(3, "Lights");
```

kaysievers commented:
Thanks a lot. That all seems to work fine.

I've updated to the current version and added 3 name entries to a real instrument: [versioduo.com/#kontra-2](https://versioduo.com/#kontra-2))

It is an acoustic MIDI instrument with two strings that produce sound. The strings can be addressed by the main MIDI device, or both strings separately played as individual MIDI instruments. The 3 devices showing up with the USB connection have descriptive names now showing in the Audio Workstation Software:

<img width="208" alt="Screenshot 2023-01-13 at 12 10 10" src="https://user-images.githubusercontent.com/1064936/212307340-205bef2f-792c-4835-8b10-c43cd660a506.png">

hathach commented:
that looks great, thank you for the extra info. We will release this shortly.
